pkg/analysis_server/lib/src/computer/computer_closingLabels.dart

```diff
@@ -37,52 +46,76 @@ class _DartUnitClosingLabelsComputerVisitor
+      // of the expression.
+      _addLabel(node, label, checkLinesUsing: node.argumentList);
     }
 
     return super.visitInstanceCreationExpression(node);
   }
 
   @override
   Object visitMethodInvocation(MethodInvocation node) {
-    if (node.argumentList != null && _spansManyLines(node)) {
+    if (node.argumentList != null) {
       final target = node.target;
       final label = target is Identifier
           ? "${target.name}.${node.methodName.name}"
           : node.methodName.name;
-      _addLabel(node, label);
+      // We override the node used for doing line calculations because otherwise methods
+      // that chain over multiple lines (but have parens on same line) would incorrectly
+      // get labels, because node.start on a methodInvocation starts at the start of the expression.
+      _addLabel(node, label, checkLinesUsing: node.argumentList);
     }
 
     return super.visitMethodInvocation(node);
   }
 
   @override
   visitListLiteral(ListLiteral node) {
-    if (_spansManyLines(node)) {
-      final args = node.typeArguments?.arguments;
-      final typeName = args != null ? args[0]?.toString() : null;
+    final args = node.typeArguments?.arguments;
+    final typeName = args != null ? args[0]?.toString() : null;
 
-      if (typeName != null) {
-        _addLabel(node, "List<$typeName>");
-      }
+    if (typeName != null) {
+      _addLabel(node, "List<$typeName>");
     }
 
     return super.visitListLiteral(node);
   }
 
-  bool _spansManyLines(AstNode node) {
-    final start = computer._lineInfo.getLocation(node.offset);
-    final end = computer._lineInfo.getLocation(node.end - 1);
-
-    return end.lineNumber - start.lineNumber > 1;
+  int interpolatedStringsEntered = 0;
+  @override
+  visitStringInterpolation(StringInterpolation node) {
+    interpolatedStringsEntered++;
+    try {
+      return super.visitStringInterpolation(node);
+    } finally {
+      interpolatedStringsEntered--;
+    }
   }
 
-  void _addLabel(AstNode node, String label) {
-    computer._closingLabels
-        .add(new ClosingLabel(node.offset, node.length, label));
+  void _addLabel(AstNode node, String label, {AstNode checkLinesUsing}) {
+    // Never add labels if we're inside strings.
+    if (interpolatedStringsEntered > 0) {
+      return;
+    }
+
+    checkLinesUsing = checkLinesUsing ?? node;
+    final start = computer._lineInfo.getLocation(checkLinesUsing.offset);
+    final end = computer._lineInfo.getLocation(checkLinesUsing.end - 1);
+    final closingLabel = new ClosingLabel(node.offset, node.length, label);
+    final labelWithSpan = new _ClosingLabelWithLineCount()
+      ..label = closingLabel
+      ..spannedLines = end.lineNumber - start.lineNumber;
+
+    if (!computer._closingLabelsByEndLine.containsKey(end.lineNumber)) {
+      computer._closingLabelsByEndLine[end.lineNumber] = [];
+    }
+    computer._closingLabelsByEndLine[end.lineNumber].add(labelWithSpan);
```

Comment:
Not necessary, but we often use a code pattern like this:
```
computer._closingLabelsByEndLine.putIfAbsent(end.lineNumber, () => <_ClosingLabelWithLineCount>[]).add(labelWithSpan);
```
which has less duplication of code. (The explicit type on the list literal is there because I don't think inference will be able to figure it out.)